src/common/iop_order.c

```diff
@@ -1016,6 +1017,24 @@ int dt_ioppr_move_iop_after(GList **_iop_list, dt_iop_module_t *module, dt_iop_m
 //--------------------------------------------------------------------
 // from here just for debug
 //--------------------------------------------------------------------
+int dt_ioppr_check_db_integrity()
+{
+  int ret = 0;
+  sqlite3_stmt *stmt;
+
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "SELECT * FROM main.history WHERE iop_order <= 0 OR iop_order IS NULL",
+                              -1, &stmt, NULL);
+  if(sqlite3_step(stmt) == SQLITE_ROW)
```

Comment:
I would go even farther. I would output the number of broken history entries and display the first 20 (columns: imgid, operation, module) or so. This could be useful to actually look at library.db and fix the issues.

What do you think?